castai/utils.go

```diff
@@ -107,3 +111,12 @@ func toNilList[T any](l *[]T) *[]T {
 	}
 	return l
 }
+
+func normalizeJSON(bytes []byte) ([]byte, error) {
```

Comment:
what is this supposed to achieve?

Reply:
Umarshal and marshal produces the same json string when the input json has differently ordered keys. Terraform would detect whitespace changes when json values are the same but their keys are ordered differently so this function normalizes the key order

Reply:
can you stress test it with a sample JSON? I think it uses maps as an intermediate holding structure, and those do not always result in same iteration order, e.g. tests based on map key order tend to be flaky for this reason.

Reply:
Added a unit test with a few test cases